packages/core/src/StateNode.ts

```diff
@@ -1661,7 +1664,7 @@ class StateNode<
     return this.resolveTransition(
       {
         configuration,
-        entrySet: [...configuration],
+        entrySet: configuration,
```

Comment:
It brings back a shared reference being used for both `entrySet` and for `configuration`. This is somewhat dangerous but I just hope that v5 algorithm works better in this regard.

It has been like that for a long time, I've "fixed" this problem [here](https://github.com/statelyai/xstate/pull/3422/files#r905127802) but that accidentally caused the mentioned regression. So now instead of copying the reference, I make sure that we don't push items to the `entrySet` that are already there:
https://github.com/statelyai/xstate/pull/3599/files#diff-7f553f4d4c285f1c05cced2e979c5b957d8884e39e74cc6797b32c63a0be7e7aR1015